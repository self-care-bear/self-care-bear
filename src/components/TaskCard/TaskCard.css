.task-card {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: grey;
  padding: 2vh;
  border-radius: 10px;
  text-transform: capitalize;
  width: 20vw;
  height: auto;
}

.task-card_title {
  margin: 1vh;
}

.task-card_buttons {
  display: flex;
  gap: 1vh;
}

.task-card_arrow {
  cursor: pointer;
  margin: 2vh 0 0 0;
}
