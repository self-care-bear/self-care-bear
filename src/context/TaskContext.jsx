import { createContext, useContext, useEffect, useState } from 'react';
import { v4 as uuid } from 'uuid';
// import { getUser } from '../services/auth';

export const TaskContext = createContext();

const TaskProvider = ({ children }) => {
  const [taskList, setTaskList] = useState([]);
  const [selectedTasks, setSelectedTasks] = useState([]);
  const value = { taskList, setTaskList, selectedTasks, setSelectedTasks };

  return <TaskContext.Provider value={value}>{children}</TaskContext.Provider>;
};

const useTasks = () => {
  const context = useContext(TaskContext);

  if (context === undefined) {
    throw new Error('To use useTasks you must wrap component in TaskProvider');
  }

  return context;
};

export { TaskProvider, useTasks };
