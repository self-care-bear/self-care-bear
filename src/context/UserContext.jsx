import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/auth';

export const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  const value = { user, setUser };

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('To use useUser you must wrap component in UserProvider');
  }

  return context;
};

export { UserProvider, useUser };
