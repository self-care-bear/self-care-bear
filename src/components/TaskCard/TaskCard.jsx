import { useState } from 'react';
import { useUser } from '../../context/UserContext';
import { deleteTask, updateTask } from '../../services/tasks';
import './TaskCard.css';

export default function TaskCard({ task, setTaskList }) {
  const { user } = useUser();
  const [isExpanded, setIsExpanded] = useState(false);
  const [isEditing, setIsEditing] = useState(false);
  const [editTask, setEditTask] = useState(task);

  const handleDelete = () => {
    deleteTask(task.id);
    setTaskList((prevState) => prevState.filter((item) => item.id !== task.id));
  };

  const handleExpand = () => {
    setIsExpanded(!isExpanded);
  };

  const handleUpdate = async (e) => {
    e.preventDefault();
    updateTask(editTask.task, editTask.task_description, user.id);
    setIsEditing(false);
  };

  return (
    <div className="task-card" onClick={handleExpand}>
      <h3>{task.task}</h3>
      {isExpanded && !isEditing && (
        <>
          <p>{task.task_description}</p>
          <button onClick={handleDelete}>delete task</button>
          <button onClick={() => setIsEditing(true)}>edit task</button>
        </>
      )}
      {isEditing && (
        <>
          <form onSubmit={handleUpdate}>
            <input
              type="text"
              value={editTask.task}
              placeholder="task"
              onChange={(e) =>
                setEditTask((prevState) => ({
                  ...prevState,
                  task: e.target.value,
                }))
              }
            />
            <input
              type="text"
              value={editTask.task_description}
              onChange={(e) =>
                setEditTask((prevState) => ({
                  ...prevState,
                  task_description: e.target.value,
                }))
              }
            />
            <button type="submit">save</button>
          </form>
        </>
      )}
    </div>
  );
}
