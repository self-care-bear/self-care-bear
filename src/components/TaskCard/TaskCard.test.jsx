import TaskCard from './TaskCard';
import { render } from '@testing-library/react';
import { UserProvider } from '../../context/UserContext';
import { TaskProvider } from '../../context/TaskContext';

test('taskcard renders', async () => {
  const { container } = render(
    <UserProvider>
      <TaskProvider>
        <TaskCard />
      </TaskProvider>
    </UserProvider>
  );

  expect(container).toMatchSnapshot();
});
