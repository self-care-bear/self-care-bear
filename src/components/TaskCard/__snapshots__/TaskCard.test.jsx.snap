// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`taskcard renders 1`] = `
<div>
  <div
    class="task-card"
  >
    <h3
      class="task-card_title"
    />
    <div
      class="task-card_buttons"
    >
      <button>
        Set Habit
      </button>
    </div>
    <p
      aria-label="expand undefined"
      class="arrow"
    >
      â–¼
    </p>
  </div>
</div>
`;
